cores/esp32/esp32-hal-bt.c

```diff
@@ -65,6 +65,11 @@ bool btStop(){
         while(esp_bt_controller_get_status() == ESP_BT_CONTROLLER_STATUS_ENABLED);
     }
     if(esp_bt_controller_get_status() == ESP_BT_CONTROLLER_STATUS_INITED){
+        if (esp_bt_controller_deinit()) {
+			log_e("BT deint failed");
+			return false;
+		}
+		while (esp_bt_controller_get_status() == ESP_BT_CONTROLLER_STATUS_INITED);
```

Comment:
waiting without delay and timeout is not a good idea. Please add delay in the while loop and also do not wait more than the time that it makes sense. Currently you will spin indefinitely and not having a delay there might not allow other tasks to run.

Reply:
Ok i have remove the while loop. In fact the state change immediately. however i've add a vTaskdelay(1) for security